### PR TITLE
Add Pin and Unpin command

### LINKED ISSUE #116: Add pin and unpin command

### PR DESCRIPTION
**Pull request by yleeyilin**
fixes #116

### CODE REVIEW

**Review by jannaleong — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/UnpinCommand.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof UnpinCommand)) {
+            return false;
+        }
+
+        return true;
+    }
```
Comment (chiageng) [inline src/main/java/seedu/address/logic/commands/UnpinCommand.java:59]: isnt that you have to check is this pin referring same Person?

File: src/main/java/seedu/address/logic/parser/PinCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * Returns name value using PREFIX.
+     * @param argMultimap Object that contains mapping of prefix to value.
+     * @return Returns object representing name.
+     * @throws ParseException Thrown when command is in invalid format.
+     */
+    public Name mapName(ArgumentMultimap argMultimap) throws ParseException {
+        try {
+            return ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(PinMessages.MESSAGE_PIN_INVALID_NAME, pe.getMessage()));
```
Comment (chiageng) [inline src/main/java/seedu/address/logic/parser/PinCommandParser.java:52]: Isnt this function same level as ParserUtil.parseName(), dont quite understand why need an additional function here

File: src/main/java/seedu/address/logic/parser/UnpinCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * @return Returns object representing name.
+     * @throws ParseException Thrown when command is in invalid format.
+     */
+    public Name mapName(ArgumentMultimap argMultimap) throws ParseException {
+        try {
+            return ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(UnpinMessages.MESSAGE_UNPIN_INVALID_NAME, pe.getMessage()));
+        }
+    }
```
Comment (chiageng) [inline src/main/java/seedu/address/logic/parser/UnpinCommandParser.java:54]: Same as PinCommandParser

File: src/main/java/seedu/address/model/person/Person.java [hunk 4]
```diff
@@ -55,10 +57,27 @@ public Email getEmail() {
...
+    }
+
+    public void toPin() {
+        pin.setPin();
+    }
+
+    public void toUnpin() {
+        pin.setUnpin();
+    }
+
```
Comment (chiageng) [inline src/main/java/seedu/address/model/person/Person.java:76]: toPin and toUnpin sounds a bit weird to me, is pin() and unpin() enough to bring out the intution that these command are actions words?

File: src/main/java/seedu/address/logic/commands/PinCommand.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof PinCommand)) {
+            return false;
+        }
+
+        return true;
+    }
+}
```
Comment (chiageng) [inline src/main/java/seedu/address/logic/commands/PinCommand.java:61]: Isnt this equal function need to change is this pin command referring to same Person?

**Review by chiageng — CHANGES_REQUESTED** [review]
Overall looks good, but i think there is some refractoring of code need to be done before merging

**Review by chiageng — APPROVED** [review]
LGTM

**Pull request merged**